docs/how-tos/xapi_transforms.rst

```diff
@@ -0,0 +1,230 @@
+.. _xapi_transforms:
+
+xAPI Transforms
+***************
+
+Aspects converts raw Open edX tracking event JSON into :ref:`xapi-concepts` for storage and analysis. This conversion process is called "transformation".
+
+Events emitted by ``openedx`` packages are transformed by `event-routing-backends`_ (ERB), a Django plugin which Aspects installs on Open edX.
+
+Transformers for events emitted by non-openedx packages should be stored close to the code that produces the events, and registered using decorators provided by `event-routing-backends`_. We
+will use OpenCraft's `completion aggregator`_ as the example for this tutorial, building on events emitted by `pr#173`_.
+
+xAPI Schema
+###########
+
+To decide on an event's xAPI schema, consider any similar events already being transformed, and what event data will be useful for analysis or visualization in Aspects.
+
+The schema for a new event must uniquely describe that event. However, it's also important to be as consistent as possible with existing event schemas so that the event can be processed and
+used in Aspects in a similar way to other events.
+
+As a reminder, an xAPI statement can be expressed as:
+
+  **Actor** **Verbed** an **Object** (within **Context**).
+
+Actor
+~~~~~
+
+For most events, the default Actor transform is enough:
+
+.. code-block:: json
+
+   {
+      "objectType": "Agent",
+      "account": {
+        "homePage": "https://lms.url",
+        "name": "32e08e30-f8ae-4ce2-94a8-c2bfe38a70cb"
+      }
+   }
+
+
+Here, the actor's `external ID`_ (of type=xapi) is used as the ``name`` field. This external ID can be matched against PII data to access the actor's name, email, and other profile details.
+
+Verb
+~~~~
+
+The verb is the primary differentiator between different xAPI events in Open edX. Select a verb that describes the event as concisely and accurately as possible, so that future, similar
+events can still be discerned.
+
+Where possible, reuse verbs from one of the registered `xAPI profiles`_. See `ERB's verb list`_ for verbs currently in use in Aspects.
```

Comment:
We should probably discourage people from reusing verbs that are already in use in ERB whenever possible, as we do a lot of processing based on the verb and it will potentially lead to confusion and maybe performance issues to overload terms.

Reply:
Talking about it, we should also update our filters to not only filter events based on the verb but also the object type and even the context activities. Mainly to allow developers to use the same verb and don't trigger any error on the downstream tables.

For the completion-aggregator and with internal libraries @andrey-canon had a lot of issues with the completion events because he was using the same verb

Reply:
> Talking about it, we should also update our filters to not only filter events based on the verb but also the object type and even the context activities. 

That's a good point.

Reply:
I've updated this PR to discourage verb re-use: https://docsopenedxorg--114.org.readthedocs.build/projects/openedx-aspects/en/114/how-tos/xapi_transforms.html#verb